src/directives/geojson.js

```diff
@@ -69,23 +70,22 @@ angular.module('ui-leaflet')
                     _remove(leafletGeoJSON);
                 };
 
-                var _addGeojson = function(model, maybeName){
-                    var geojson = angular.copy(model);
+                var _addGeojson = function(geojson, maybeName){
+
                     if (!(isDefined(geojson) && isDefined(geojson.data))) {
                         return;
                     }
                     var onEachFeature = _hookUpEvents(geojson, maybeName);
 
                     if (!isDefined(geojson.options)) {
-                        //right here is why we use a clone / copy (we modify and thus)
-                        //would kick of a watcher.. we need to be more careful everywhere
-                        //for stuff like this
-                        geojson.options = {
-                            style: geojson.style,
-                            filter: geojson.filter,
-                            onEachFeature: onEachFeature,
-                            pointToLayer: geojson.pointToLayer
-                        };
+                        hlp.modelChangeInDirective(watchTrap, "changeFromDirective", () => {
```

Comment:
If no options are defined then we add options (as a convenience). But we must tell the directive to not use the watch!